netlab/L5/tls_protocol_layer.hpp

```diff
@@ -556,30 +559,34 @@ namespace netlab {
 				it++;
 				this->handshake.body.serverHello.compression_method = static_cast<CompressionMethod>(*it);
 				// Extensions
+				//break;
+				it++;
 				extensions_length = *it;
 				it++;
 				if (extensions_length > 0) {
 					this->handshake.body.serverHello.extensions_present = true;
-					for (uint16_t i = 0; i < extensions_length; i++) {
+					for (uint16_t i = 0; i < extensions_length; ) {
 						// Read the Extension
-						Extension extension(it, it + sizeof(Extension));
+						Extension extension(it, it + extensions_length);
 						this->handshake.body.serverHello.extensions_union.extensions.push_back(extension);
-						it += sizeof(Extension);
+						it += extensions_length;
+						i += extensions_length;
 					}
 				}
 				break;
 			case CERTIFICATE:
 				// Deserialize certificate struct
 				certificate_list_length = deserialize_3_bytes(it);
 				this->handshake.body.certificate.certificate_list.resize(certificate_list_length - CERTIFICATE_HANDSHAKE_OFFSET_LENGTH);
-				for (size_t i = 0; i < certificate_list_length - CERTIFICATE_HANDSHAKE_OFFSET_LENGTH; i++) {
+				for (size_t i = 0; i < certificate_list_length - CERTIFICATE_HANDSHAKE_OFFSET_LENGTH; ) {
```

Comment:
Just curious about the difference between incrementing i in the for loop statement in comparison to this style

Reply:
you dont know how much to increament, cartificates are not fixed in size